File: U1/EXAMEN_SOAP_DOTNET_GR01/01 SERVIDOR/BanquitoServer_Soap_DotNet_GR01/BanquitoServer_Soap_DotNet_GR01/Validators/SolicitudCreditoValidator.cs
using System;
using System.Globalization;
using System.Text.RegularExpressions;
using BanquitoServer_Soap_DotNet_GR01.Constants;
using BanquitoServer_Soap_DotNet_GR01.DTOs;

namespace BanquitoServer_Soap_DotNet_GR01.Validators
{
    /// <summary>
    /// Validador para SolicitudCreditoDTO
    /// </summary>
    public static class SolicitudCreditoValidator
    {
        /// <summary>
        /// Validar una solicitud de crédito y convertir valores
        /// </summary>
        /// <param name="solicitud">Solicitud a validar</param>
        /// <param name="precio">Precio convertido (out)</param>
        /// <param name="cuotas">Número de cuotas convertido (out)</param>
        /// <param name="mensajeError">Mensaje de error si la validación falla</param>
        /// <returns>True si es válida, False si no</returns>
        public static bool ValidarYConvertir(SolicitudCreditoDTO solicitud, out decimal precio, out int cuotas, out string mensajeError)
        {
            precio = 0;
            cuotas = 0;
            mensajeError = string.Empty;

            // Validar que la solicitud no sea nula
            if (solicitud == null)
            {
                mensajeError = ErrorMessages.SolicitudNula;
                return false;
            }

            // Validar cédula no vacía
            if (string.IsNullOrWhiteSpace(solicitud.Cedula))
            {
                mensajeError = ErrorMessages.CedulaRequerida;
                return false;
            }

            // Validar formato de cédula (10 dígitos)
            if (!Regex.IsMatch(solicitud.Cedula, @"^\d{10}$"))
            {
                mensajeError = ErrorMessages.CedulaInvalida;
                return false;
            }

            // Validar y convertir precio del electrodoméstico
            if (string.IsNullOrWhiteSpace(solicitud.PrecioElectrodomestico))
            {
                mensajeError = ErrorMessages.PrecioInvalido;
                return false;
            }

            if (!decimal.TryParse(solicitud.PrecioElectrodomestico, NumberStyles.Any, CultureInfo.InvariantCulture, out precio))
            {
                mensajeError = ErrorMessages.PrecioInvalido;
                return false;
            }

            if (precio <= 0)
            {
                mensajeError = ErrorMessages.PrecioInvalido;
                return false;
            }

            // Validar y convertir número de cuotas
            if (string.IsNullOrWhiteSpace(solicitud.NumeroCuotas))
            {
                mensajeError = ErrorMessages.NumeroCuotasInvalido;
                return false;
            }

            if (!int.TryParse(solicitud.NumeroCuotas, out cuotas))
            {
                mensajeError = ErrorMessages.NumeroCuotasInvalido;
                return false;
            }

            if (cuotas <= 0)
            {
                mensajeError = ErrorMessages.NumeroCuotasInvalido;
                return false;
            }

            return true;
        }
    }
}

File: U2/PROYECTO/BanquitoServer_Restfull_Java_GR01/src/main/java/ec/edu/monster/Validators/SolicitudCreditoValidator.java
package ec.edu.monster.Validators;

import ec.edu.monster.Constants.ErrorMessages;
import ec.edu.monster.DTOs.SolicitudCreditoDTO;
import java.math.BigDecimal;
import java.util.regex.Pattern;

/**
 * Validador para SolicitudCreditoDTO
 */
public final class SolicitudCreditoValidator {

    private static final Pattern CEDULA_PATTERN = Pattern.compile("^\\d{10}$");

    private SolicitudCreditoValidator() {
    }

    /**
     * Validar una solicitud de crédito y convertir valores
     */
    public static ValidationResult validarYConvertir(SolicitudCreditoDTO solicitud) {
        if (solicitud == null) {
            return new ValidationResult(false, ErrorMessages.SOLICITUD_NULA, null, null);
        }

        // Validar cédula no vacía
        if (solicitud.getCedula() == null || solicitud.getCedula().isBlank()) {
            return new ValidationResult(false, ErrorMessages.CEDULA_REQUERIDA, null, null);
        }

        // Validar formato de cédula (10 dígitos)
        if (!CEDULA_PATTERN.matcher(solicitud.getCedula()).matches()) {
            return new ValidationResult(false, ErrorMessages.CEDULA_INVALIDA, null, null);
        }

        // Validar y convertir precio del electrodoméstico
        if (solicitud.getPrecioElectrodomestico() == null || solicitud.getPrecioElectrodomestico().isBlank()) {
            return new ValidationResult(false, ErrorMessages.PRECIO_INVALIDO, null, null);
        }

        BigDecimal precio;
        try {
            precio = new BigDecimal(solicitud.getPrecioElectrodomestico());
        } catch (NumberFormatException e) {
            return new ValidationResult(false, ErrorMessages.PRECIO_NO_NUMERICO, null, null);
        }

        if (precio.compareTo(BigDecimal.ZERO) <= 0) {
            return new ValidationResult(false, ErrorMessages.PRECIO_INVALIDO, null, null);
        }

        // Validar y convertir número de cuotas
        if (solicitud.getNumeroCuotas() == null || solicitud.getNumeroCuotas().isBlank()) {
            return new ValidationResult(false, ErrorMessages.NUMERO_CUOTAS_INVALIDO, null, null);
        }

        Integer cuotas;
        try {
            cuotas = Integer.parseInt(solicitud.getNumeroCuotas());
        } catch (NumberFormatException e) {
            return new ValidationResult(false, ErrorMessages.NUMERO_CUOTAS_NO_NUMERICO, null, null);
        }

        if (cuotas <= 0) {
            return new ValidationResult(false, ErrorMessages.NUMERO_CUOTAS_INVALIDO, null, null);
        }

        return new ValidationResult(true, null, precio, cuotas);
    }

    /**
     * Clase para resultado de validación con valores convertidos
     */
    public static class ValidationResult {
        private final boolean valido;
        private final String mensajeError;
        private final BigDecimal precio;
        private final Integer cuotas;

        public ValidationResult(boolean valido, String mensajeError, BigDecimal precio, Integer cuotas) {
            this.valido = valido;
            this.mensajeError = mensajeError;
            this.precio = precio;
            this.cuotas = cuotas;
        }

        public boolean isValido() {
            return valido;
        }

        public String getMensajeError() {
            return mensajeError;
        }

        public BigDecimal getPrecio() {
            return precio;
        }

        public Integer getCuotas() {
            return cuotas;
        }
    }
}
